azurerm/resource_arm_automation_dsc_configuration.go

```diff
@@ -0,0 +1,141 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscConfigurationRead,
+		Update: resourceArmAutomationDscConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"account_name": {
```

Comment:
I would probably be explicit here and call this automation_account_name

Reply:
Agreed, this should be `automation_account_name`

Reply:
I was following already established Automation related resources like [credential](https://www.terraform.io/docs/providers/azurerm/r/automation_credential.html)

```tf
resource "azurerm_automation_credential" "example" {
  name                = "credential1"
  resource_group_name = "${azurerm_resource_group.example.name}"
  account_name        = "${azurerm_automation_account.example.name}"
  username           = "example_user"
  password            = "example_pwd"
  description         = "This is an example credential"
}
```

Do I still need to change mine (it will result in inconsistent parameter naming)?

Reply:
Ah. Makes me slightly sad but I'd probably say consistency over correctness :cry: @katbyte your call.

Reply:
Or we could deprecate the account_name in azurerm_automation_credential.

Reply:
Please do, the other resources are incorrect and will need to be changed eventually.

As much as I would usually agree in consistency over correctness, this isn't a very confusing change so for a new resource I'm going to go with its better to use the correct name.

I have already depreciated `account_name` in `azurerm_automation_schedule`, so feel free to apply that to the rest of the automation resources, preferably in a separate PR 🙂